src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClientDiscovery.cs

```diff
@@ -229,6 +236,40 @@ async Task<ResolveResult[]> VerifyContacts()
         }
     }
 
+    /// <summary>
+    /// Trim the number of Contact in the `fullContact` array to `count` length
+    /// by picking random elements while avoiding repeating elements from being returned
+    /// </summary>
+    /// <param name="fullContact">Array of Contacts</param>
+    /// <param name="count">The number of elements to return</param>
+    /// <returns></returns>
+    private static Contact[] TrimContacts(Contact[] fullContact, int count)
+    {
+        if (fullContact.Length <= count)
+            return fullContact;
+        
+        Shuffle(fullContact);
+        return fullContact.Take(count).ToArray();
+    }
+
+    /// <summary>
+    /// Fisher-Yates in-place array shuffle algorithm
+    /// </summary>
+    /// <param name="array"></param>
+    /// <typeparam name="T"></typeparam>
+    private static void Shuffle<T>(T[] array)
```

Comment:
Do we have another implementation of this lying around somewhere? I could have sworn I'd written one before

Reply:
no idea

Reply:
There is one in the Akka.Util.Internal.ArrayExtensions, but its internal

Reply:
I would add a link to that file and just use that (or you can add a friend assembly to Akka.Discovery; link is probably cleaner.)

Reply:
That's just a nitpick though

Reply:
Done, turns out the core Akka project is already a friend of Akka.Cluster.Tools